docs/docsite/rst/user_guide/playbooks_variables.rst

```diff
@@ -336,11 +341,15 @@ Ansible merges different variables set in inventory so that more specific settin
 
 .. rubric:: Footnotes
 
-.. [1] Tasks in each role see their own role's defaults. Tasks defined outside of a role see the last role's defaults.
+.. [1] Tasks in each role see their own role's defaults if specified, or those from the role's dependencies. Tasks defined outside of a role see the last role's defaults.
 .. [2] Variables defined in inventory file or provided by dynamic inventory.
 .. [3] Includes vars added by 'vars plugins' as well as host_vars and group_vars which are added by the default vars plugin shipped with Ansible.
-.. [4] When created with set_facts's cacheable option, variables have the high precedence in the play,
-       but are the same as a host facts precedence when they come from the cache.
+.. [4] When created with set_facts's cacheable option, variables have the high precedence in the play, but are the same as a host facts precedence when they come from the cache.
+.. [5] Tasks in each role see their own role's vars if specified, or those from the role's dependencies. Tasks defined outside of a role see the last role's vars.
+.. [6] Tasks in each role see their own role's params if specified, or those from the role's dependencies. Role params are not exposed outside of the role.
+.. [7] Role params are not specified within ``vars:`` they are top level params specified as part of the role definition
+.. [8] Roles loaded using ``include_role`` do not have their defaults or vars available in the play scope until they are encountered during the playbook execution, and when ``public: true`` is defined. Use of ``roles:`` or ``import_role`` makes the defaults and vars available to the play scope at playbook parsing time.
```

Comment:
This is useful information, could we add a sentence or note about what difference this makes for the user? Do role vars for tasks in a role hold a different "rank" in variable precedence when the role is loaded with `include_role` because those vars haven't been loaded yet when the main comparison is made? If so, is the precedence rank lower or higher than when the role is loaded with `import_role` or just in a `roles:` list?